ports/rp2/machine_adc.c

```diff
@@ -78,7 +78,10 @@ STATIC mp_obj_t machine_adc_make_new(const mp_obj_type_t *type, size_t n_args, s
         }
     }
 
-    adc_init();
+    // Initialise the ADC peripheral if it's not already running.
+    if (!(adc_hw->cs & ADC_CS_READY_BITS)) {
```

Comment:
Having had a quick look at the RP2040 datasheet (and the SDK source), I think this should be checking `ADC_CS_EN_BITS` rather than `ADC_CS_READY_BITS` (see [adc_init](https://github.com/raspberrypi/pico-sdk/blob/master/src/rp2_common/hardware_adc/adc.c#L11) ) ?
(`ADC_CS_READY_BITS` is already checked by [adc_read](https://github.com/raspberrypi/pico-sdk/blob/master/src/rp2_common/hardware_adc/include/hardware/adc.h#L124) )

Reply:
I checked the READY_BITS because that's what adc_init finishes with, a test that they are enabled.  But I did originally consider using EN_BITS because they do seem more logical as a test of the peripheral enabled.  I can change it.

Reply:
Updated.  Tested.